### PR TITLE
Feature - search request table

### PR DESCRIPTION
**Pull request by yonikid15**
# Notes
- Creates table for search requests
- Adds graphql operation to fetch all search requests
- Creates search requests routes 
- Adds search request page to dashboard

### CODE REVIEW

File: admin/resources/js/adminRoutes.ts [hunk 1]
```diff
@@ -47,3 +50,8 @@ export const userTablePath = (): string => `/${ADMIN_APP_DIR}/users`;
 export const userCreatePath = (): string => `/${ADMIN_APP_DIR}/users/create`;
 export const userUpdatePath = (id: string): string =>
   `/${ADMIN_APP_DIR}/users/${id}/edit`;
+
+export const searchRequestTablePath = (): string =>
+  `/${ADMIN_APP_DIR}/searchRequests`;
```
Comment (mnigh) [inline admin/resources/js/adminRoutes.ts:55]:
```suggestion
  `/${ADMIN_APP_DIR}/search-requests`;
```

File: admin/resources/js/adminRoutes.ts [hunk 2]
```diff
@@ -47,3 +50,8 @@ export const userTablePath = (): string => `/${ADMIN_APP_DIR}/users`;
...
+export const searchRequestUpdatePath = (id: string): string =>
+  `/${ADMIN_APP_DIR}/searchRequests/${id}/edit`;
```
Comment (mnigh) [inline admin/resources/js/adminRoutes.ts:57]:
```suggestion
  `/${ADMIN_APP_DIR}/search-requests/${id}/edit`;
```

File: admin/resources/js/components/PoolDashboard.tsx [hunk 3]
```diff
@@ -26,6 +26,8 @@ import {
   userUpdatePath,
   homePath,
   homePublicPath,
+  searchRequestTablePath,
+  searchRequestUpdatePath,
```
Comment (mnigh) [inline admin/resources/js/components/PoolDashboard.tsx:30]: 'searchRequestUpdatePath' is declared but its value is never read.

File: admin/resources/js/components/PoolDashboard.tsx [hunk 4]
```diff
@@ -227,12 +230,33 @@ const routes: Routes<RouterResult> = [
...
+  },
 ];
 
 export const PoolDashboard: React.FC = () => {
   const intl = useIntl();
 
   const menuItems = [
+    <MenuHeading
+      key="search-requests"
+      text={intl.formatMessage({
```
Comment (mnigh) [inline admin/resources/js/components/PoolDashboard.tsx:247]: Is there a reason this isn't in the `messages` const on line 60?

File: admin/resources/js/components/PoolDashboard.tsx [hunk 5]
```diff
@@ -227,12 +230,33 @@ const routes: Routes<RouterResult> = [
...
+        defaultMessage: "Requests",
+        description: "Label displayed on the requests menu item.",
+      })}
+    />,
+    <MenuLink
+      key="all-requests"
+      href={searchRequestTablePath()}
+      text={intl.formatMessage({
```
Comment (mnigh) [inline admin/resources/js/components/PoolDashboard.tsx:255]: Is there a reason this isn't in the `messages` const on line 60?

File: admin/resources/js/components/searchRequest/SearchRequestTable.tsx [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+       Header: intl.formatMessage({
+        defaultMessage: "Status",
+        description: "Title displayed on the search request table status column.",
+      }),
+       accessor: "status"
+     },
+     {
+       Header: intl.formatMessage({
+        defaultMessage: "Manager",
+        description: "Title displayed on the search request table first name column.",
```
Comment (mnigh) [inline admin/resources/js/components/searchRequest/SearchRequestTable.tsx:47]:
```suggestion
        description: "Title displayed on the search request table manager column.",
```

File: admin/resources/js/components/searchRequest/SearchRequestTable.tsx [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+ );
+
+ const memoizedData = useMemo(
+   () => poolCandidateSearchRequests.filter(notEmpty),
+   [poolCandidateSearchRequests],
+ );
+
+ return (
+   <>
+    <Table data={memoizedData} columns={columns} />
```
Comment (mnigh) [inline admin/resources/js/components/searchRequest/SearchRequestTable.tsx:111]:
```suggestion
    <Table data={memoizedData} columns={columns} hiddenCols={["id"]} />
```

File: admin/resources/js/components/searchRequest/SearchRequestTable.tsx [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+        description: "Title displayed on the search request table job title column.",
+      }),
+       accessor: "jobTitle"
+     },
+     {
+       Header: intl.formatMessage({
+        defaultMessage: "Pool",
+        description: "Title displayed on the search request table pool column.",
+      }),
+       accessor: ({ poolCandidateFilter }) => poolCandidateFilter.id
```
Comment (mnigh) [inline admin/resources/js/components/searchRequest/SearchRequestTable.tsx:77]: Is this meant to be the name (in EN or FR) of the Pool?

File: admin/resources/js/stories/SearchRequest.stories.tsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import { storiesOf } from "@storybook/react";
+import { fakeSearchRequests } from "@common/fakeData";
+import { SearchRequestTable } from "../components/searchRequest/SearchRequestTable";
+
+// It is possible data may come back from api with missing data.
```
Comment (mnigh) [inline admin/resources/js/stories/SearchRequest.stories.tsx:6]:
```suggestion
```

File: admin/resources/js/stories/SearchRequest.stories.tsx [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+const stories = storiesOf("Search Results", module);
+
+stories.add("Search Result Table", () => (
```
Comment (mnigh) [inline admin/resources/js/stories/SearchRequest.stories.tsx:10]:
```suggestion
stories.add("Search Results Table", () => (
```

**Review by mnigh — CHANGES_REQUESTED** [review]
Nice work! Some questions and concerns to address before it is ready to be approved, though.

File: admin/resources/js/components/PoolDashboard.tsx (hunk 5 above)
Comment (yonikid15) [reply admin/resources/js/components/PoolDashboard.tsx:255]: Oh ya, so should I move each message, in messages, to where they're being referenced.. or just move these into messages? I think the former.

File: admin/resources/js/components/searchRequest/SearchRequestTable.tsx (hunk 8 above)
Comment (yonikid15) [reply admin/resources/js/components/searchRequest/SearchRequestTable.tsx:77]: Yea it is

File: admin/resources/js/stories/SearchRequest.stories.tsx [hunk 11]
```diff
@@ -3,11 +3,10 @@ import { storiesOf } from "@storybook/react";
 import { fakeSearchRequests } from "@common/fakeData";
 import { SearchRequestTable } from "../components/searchRequest/SearchRequestTable";
 
-// It is possible data may come back from api with missing data.
```
Comment (mnigh) [inline admin/resources/js/stories/SearchRequest.stories.tsx:6]:
```suggestion
```

**Review by mnigh — CHANGES_REQUESTED** [review]
Nice work! I especially like the linked pool names in the _All Requests_ table. A few things before approval:

- One small extra space should be removed
- A few npm scripts should be run and the `generated.ts` and `package-lock.json` files committed, see screenshot:

Screen Shot 2021-11-09 at 11 34 38
.

**Review by mnigh — APPROVED** [review]
👍

**Pull request merged**